test/unit/services/CancellationXmlValidationServiceSpec.scala

```diff
@@ -0,0 +1,116 @@
+/*
```

Comment:
Cancellation and Submission are subclasses of XmlValidationService - they have only one minor difference which is `location` . So I would expect only one big `Spec` - with possibly two subclass `Spec` .